crd-generator/cli/src/main/java/io/fabric8/crd/generator/cli/CRDGeneratorExecutionExceptionHandler.java

```diff
@@ -20,14 +20,16 @@
 import org.slf4j.LoggerFactory;
 import picocli.CommandLine;
 
+import java.util.function.Supplier;
+
 class CRDGeneratorExecutionExceptionHandler implements CommandLine.IExecutionExceptionHandler {
 
   private static final Logger log = LoggerFactory.getLogger(CRDGeneratorExecutionExceptionHandler.class);
 
-  private final CRDGeneratorCLI crdGeneratorCLI;
+  private final Supplier<String> diagTextSupplier;
```

Comment:
nitpick: maybe record the result of calling the supplier instead of calling the supplier repeatedly?

Reply:
The diag text is not yet available when the constructor of CRDGeneratorExecutionExceptionHandler is called. So I have to fetch it somehow later if an error occurs from the CLI class. To prevent a circular reference it's a supplier, so technically cheating :smirk:. But anyways I think it will be called at most only once: each if statement has a return statement.